From c257243882c2f6ab3ceaea7ec99a462f53fb0e1c Mon Sep 17 00:00:00 2001
From: Boyuan Yang <byang@debian.org>
Date: Sun, 30 Oct 2022 00:50:55 -0400
Subject: [PATCH] debian/apparmor/: Allow reading disable_ipv6 file under /proc

Otherwise syslog is spammed by repeated AppArmor denial:

    apparmor="DENIED" operation="open" profile="/usr/sbin/cups-browsed"
    name="/proc/sys/net/ipv6/conf/all/disable_ipv6" pid=61278
    comm="cups-browsed" requested_mask="r" denied_mask="r" fsuid=0 ouid=0

This patch would Closes: #1019270.
---
 debian/apparmor/usr.sbin.cups-browsed | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/debian/apparmor/usr.sbin.cups-browsed b/debian/apparmor/usr.sbin.cups-browsed
index 9c2dbff..cab1cd0 100644
--- a/debian/apparmor/usr.sbin.cups-browsed
+++ b/debian/apparmor/usr.sbin.cups-browsed
@@ -17,6 +17,9 @@
   /var/log/cups/* rw,
   /tmp/** rw,
 
+  # Allow reading system ipv6 status
+  /proc/sys/net/ipv6/conf/all/disable_ipv6 r,
+
   # Courtesy to the Debian Edu team...
   /etc/cups/cups-browsed-debian-edu.conf r,
 
-- 
2.37.2

